### ISSUE TITLE
Queue was stuck until I updated "scheduled_at" on one worker

### ISSUE THREAD

**Issue by darksheik** [sorentwo/oban#417]
### Environment

Oban Version: 2.4.2
PostgreSQL Version: 11.7 (Aurora)
Elixir & Erlang/OTP Versions (`elixir --version`)
```
Erlang/OTP 23 [erts-11.0.2] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]

Elixir 1.10.4 (compiled with Erlang/OTP 23)
```

### Current Behavior

I wanted to report some strange behavior I just produced by accident - it doesn't really make sense logically but it might to you.

I had a bunch of available jobs in a queue and it wasn't picking up any.  Usually I attribute this to a stuck worker after a hot deploy, and scaling up usually creates more slots and jobs will be executed in those new slots.  But in this case, it didn't.  I did a full restart and my queue of about 27,000 _still_ wasn't moving.  And it was only this one queue - other queues were picking up jobs fine.

Some of the jobs had been in the queue for a couple of days as we've been catching up.  For a goof, I decided to update the `scheduled_at` one of the jobs to see if that would cause that one job to be picked up.  To my surprise, just changing the value on that one job caused the entire queue to start moving again.  Did this perhaps execute a trigger that had the side effect of making the queue move again?

```
begin;
BEGIN
update oban_jobs set scheduled_at = now() where id = 96692840;
UPDATE 1
commit;
COMMIT
```

I do have a snapshot of the worker and I notice that on this particular record, `completed_at` and `discarded_at` were set even though the job was in an `available` state.  Wondering if that's a clue as well.  Providing other values here which might be relevant.

```
  attempt: 5,
  attempted_at: ~U[2021-01-28 19:10:52.551209Z],
  cancelled_at: nil,
  completed_at: ~U[2021-01-28 19:12:07.624920Z],
  discarded_at: ~U[2021-01-28 19:12:34.975874Z],
  id: 96692840,
  inserted_at: ~U[2021-01-28 19:00:42.807974Z],
  max_attempts: 5,
  meta: %{},
  priority: 0,
  replace_args: nil,
  scheduled_at: ~U[2021-01-28 19:12:54.177066Z],
  state: "available",
  tags: [],
  unique: nil,
  unsaved_error: nil
```

### Expected Behavior

Jobs in available queue should be picked up.

**Comment by sorentwo** [comment]
tl;dr — this is a one off, if you've upgraded Oban Web to v2.5.1 it won't happen again

This was _very_ likely caused by retrying a job that was previously discarded because the attempts were exhausted.

Prior to v2.5.1, the retry mechanism in Oban Web wouldn't always bump the `max_attempts` on retry. That could to a situation where attempts > max_attempts, which shouldn't ever happen and is now prevented by a table level check constraint. When that particular queue went to fetch new jobs it tried to bump the `attempt` for that job, raise a db error, and that would trip the circuit breaker. The queue would then wait for the breaker to open, try again, and hit the same issue.

If you haven't already, I highly recommend hooking into [circuit events](https://hexdocs.pm/oban/Oban.Telemetry.html#module-circuit-events) and reporting errors.
